File: packages/sapplauz_core/lib/models/update_user_request.dart
class UpdateUserRequest {
  final String firstName;
  final String lastName;
  final String email;
  final String? currentPassword;
  final String? newPassword;

  UpdateUserRequest({
    required this.firstName,
    required this.lastName,
    required this.email,
    this.currentPassword,
    this.newPassword,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
    };
    
    if (currentPassword != null && currentPassword!.isNotEmpty) {
      json['currentPassword'] = currentPassword!;
    }
    
    if (newPassword != null && newPassword!.isNotEmpty) {
      json['newPassword'] = newPassword!;
    }
    
    return json;
  }
}

File: backend/SApplauz.Shared/DTOs/UpdateUserRequest.cs
namespace SApplauz.Shared.DTOs;

public class UpdateUserRequest
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? CurrentPassword { get; set; } // Required only if changing password
    public string? NewPassword { get; set; } // Optional - only if user wants to change password
}






